runtime/src/accounts_db.rs

```diff
@@ -2103,4 +2132,56 @@ pub mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_hash_stored_account() {
+        // This test uses some UNSAFE trick to detect most of account's field
+        // addition and deletion without changing the hash code
+
+        const ACCOUNT_DATA_LEN: usize = 3;
+        // the type of InputTuple elements must not contain references;
+        // they should be simple scalars or data blobs
+        type InputTuple = (
+            Slot,
+            StoredMeta,
+            AccountMeta,
+            [u8; ACCOUNT_DATA_LEN],
+            usize, // for StoredAccount::offset
+            Hash,
+        );
+        const INPUT_LEN: usize = std::mem::size_of::<InputTuple>();
+        type InputBlob = [u8; INPUT_LEN];
+        let mut blob: InputBlob = [0u8; INPUT_LEN];
+
+        // spray memory with decreasing counts so that, data layout can be detected.
+        for (i, byte) in blob.iter_mut().enumerate() {
+            *byte = (INPUT_LEN - i) as u8;
+        }
+
+        //UNSAFE: forcibly cast the special byte pattern to actual account fields.
```

Comment:
My first UNSAFE here. :) Is there any better way to archive similar test result?

Reply:
Neat idea.  An `unsafe` in a test is less concerning to me

Reply:
> An unsafe in a test is less concerning to me

Thx! This will be a precedent for the follow-up PRs, in which we need more of these crafted binaries to protect validator from crashing from external data. :)